src/async-wrap.cc

```diff
@@ -407,6 +407,47 @@ static void DisablePromiseHook(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+class DestroyParam {
+ public:
+  double asyncId;
+  v8::Persistent<Object> target;
+  v8::Persistent<Object> propBag;
+};
+
+
+void AsyncWrap::WeakCallback(
+  const v8::WeakCallbackInfo<DestroyParam>& info
+) {
+  HandleScope scope(info.GetIsolate());
+
+  Environment* env = Environment::GetCurrent(info.GetIsolate());
+  DestroyParam* p = info.GetParameter();
+  Local<Object> propBag = PersistentToLocal(info.GetIsolate(), p->propBag);
+
+  Local<Value> val = propBag->Get(env->destroyed_string());
+  if (!val->BooleanValue()) {
+    PushBackDestroyAsyncId(env, p->asyncId);
+  }
+  p->propBag.Reset();
```

Comment:
Can you maybe `CHECK(p->target.IsEmpty());` too, just to be sure?

Reply:
That fails. I'm pretty sure the pointer just becomes invalid, but still points somewhere.

Reply:
Hm, in that case, I would *assume* the pointer is still valid – but either way, calling `.Reset()` on it should be fine?

Reply:
Hm, in that case, I would *assume* the pointer is still valid – but either way, calling `.Reset()` on it should be fine?

Reply:
Oh, that works. I was pretty sure that failed previously, but owell, apparently misremembering.